slackclient/server.py

```diff
@@ -70,6 +84,29 @@ def append_user_agent(self, name, version):
     def rtm_connect(self, reconnect=False, timeout=None, use_rtm_start=True, **kwargs):
         # rtm.start returns user and channel info, rtm.connect does not.
         connect_method = "rtm.start" if use_rtm_start else "rtm.connect"
+
+        # If the `auto_reconnect` param was passed, set the server's `auto_reconnect` attr
+        if kwargs and kwargs["auto_reconnect"] is True:
+            self.auto_reconnect = True
+
+        # If this is an auto reconnect, rate limit reconnect attempts
+        if self.auto_reconnect and reconnect:
+            # Raise a SlackConnectionError after 5 retries within 3 minutes
+            if self.reconnect_attempt == 5:
```

Comment:
This feels like a value that shouldn't be hidden down here in the guts?

Reply:
I agree, but I'm not quite sure where else it could go 🤔